pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -76,6 +77,13 @@ func (srv PrometheusSrv) RouteGetRuleStatuses(c *models.ReqContext) response.Res
 			continue
 		}
 		groupId, namespaceUID, namespace := r[0], r[1], r[2]
+		if _, err := srv.store.GetNamespaceByUID(namespaceUID, c.SignedInUser.OrgId, c.SignedInUser); err != nil {
```

Comment:
If you have a 200 folders won't this result in N+ queries ?

Reply:
yes, as an improvement we could use some caching but we need to cache also folder permissions and keep the cache updated whenever permissions change.